server/src/test/java/e2e/WorkflowEventsTest.java

```diff
@@ -1,31 +1,94 @@
 package e2e;
 
+import io.littlehorse.sdk.common.proto.AwaitWorkflowEventRequest;
+import io.littlehorse.sdk.common.proto.LHStatus;
+import io.littlehorse.sdk.common.proto.LittleHorseGrpc.LittleHorseBlockingStub;
+import io.littlehorse.sdk.common.proto.PutWorkflowEventDefRequest;
+import io.littlehorse.sdk.common.proto.RunWfRequest;
+import io.littlehorse.sdk.common.proto.VariableType;
+import io.littlehorse.sdk.common.proto.WfRunId;
+import io.littlehorse.sdk.common.proto.WorkflowEvent;
+import io.littlehorse.sdk.common.proto.WorkflowEventDefId;
+import io.littlehorse.sdk.wfsdk.WfRunVariable;
 import io.littlehorse.sdk.wfsdk.Workflow;
 import io.littlehorse.sdk.wfsdk.internal.WorkflowImpl;
 import io.littlehorse.test.LHTest;
 import io.littlehorse.test.LHWorkflow;
+import io.littlehorse.test.LHWorkflowEvent;
 import io.littlehorse.test.WorkflowVerifier;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.MethodOrderer;
+import org.junit.jupiter.api.Order;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestMethodOrder;
 
 @LHTest
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
 public class WorkflowEventsTest {
 
     @LHWorkflow("events")
     private Workflow eventsWf;
 
+    private LittleHorseBlockingStub client;
     private WorkflowVerifier verifier;
 
     @Test
-    public void shouldDoBasic() {
-        // Commented out until we add the ability to create a WorkflowEventDef to the test
-        // framework.
-        // verifier.prepareRun(eventsWf).waitForStatus(LHStatus.COMPLETED).start();
+    @Order(1)
+    void shouldBeAbleToGetWorkflowEventThrownFirst() {
+        verifier.prepareRun(eventsWf)
+                .waitForStatus(LHStatus.COMPLETED)
+                .thenVerifyWfRun(wfRun -> {
+                    WorkflowEvent result = client.withDeadlineAfter(1, TimeUnit.SECONDS)
+                            .awaitWorkflowEvent(AwaitWorkflowEventRequest.newBuilder()
+                                    .setWfRunId(wfRun.getId())
+                                    .addEventDefIds(
+                                            WorkflowEventDefId.newBuilder().setName("user-created"))
+                                    .build());
+
+                    Assertions.assertThat(result.getContent().getStr()).isEqualTo("hello there");
+                })
+                .start();
+    }
+
+    @Test
+    @Order(2) // has to go after the first one, which registers the WfSpec
+    void shouldBeAbleToGetWorkflowEventThrownAfterRequest() {
+        WfRunId id = WfRunId.newBuilder().setId(UUID.randomUUID().toString()).build();
```

Comment:
```suggestion
        WfRunId id = WfRunId.newBuilder().setId(UUID.randomUUID().toString()).build();
        // Delay the WfRun so that the rpc AwaitWorkflowEvent gets there before the WorkflowEvent is thrown.
        ScheduledExecutorService scheduledExecutorService = Executors.newScheduledThreadPool(2);
        scheduledExecutorService.submit(() -> {
            WorkflowEvent event = client.awaitWorkflowEvent(AwaitWorkflowEventRequest.newBuilder()
                    .setWfRunId(id)
                    .addEventDefIds(WorkflowEventDefId.newBuilder().setName("user-created"))
                    .build());
            Assertions.assertThat(event.getContent().getStr()).isEqualTo("hello there");
        });
        scheduledExecutorService.schedule(() -> verifier.prepareRun(eventsWf).start(id), 1, TimeUnit.SECONDS);
```

I think this is what you want

Reply:
Then wait for termination of course.

Reply:
Got it, your proposal tests the exact same thing. It's not going to ever be flakey with either your implementation or mine, so I'm not too worried about that.

But perhaps yours is easier to understand :smile: I didn't know that `ScheduledExecutorService` existed.